staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -240,3 +250,12 @@ func (r *proxyHandler) updateAPIService(apiService *apiregistrationapi.APIServic
 	}
 	r.handlingInfo.Store(newInfo)
 }
+
+func (r *proxyHandler) updateAPIServiceStatus(apiService *apiregistrationapi.APIService) {
+	value := r.handlingInfo.Load()
+	if value != nil {
+		handlingInfo := value.(proxyHandlingInfo)
+		handlingInfo.serviceAvailable = apiregistrationapi.IsAPIServiceConditionTrue(apiService, apiregistrationapi.Available)
```

Comment:
don't have to clone to avoid a data race?

Reply:
yes, thx.